doc/getting-started.rst

```diff
@@ -1,90 +1,100 @@
-Getting Started with Haskell and Cabal
-======================================
+Getting Started
+===============
 
-Installing the Haskell toolchain
---------------------------------
-
-To install the Haskell toolchain follow the `ghcup instructions
-<https://www.haskell.org/ghcup/>`__.
+Installing Cabal
+----------------
 
+The easiest and recommended way to install the ``cabal`` commandline tool 
+on Linux, macOS, FreeBSD or Windows is through `ghcup <https://www.haskell.org/ghcup/>`__. 
+It installs the "Haskell toolchain" which includes Cabal,
+the Haskell compiler `GHC <https://www.haskell.org/ghc/>`__ 
+and optionally other useful Haskell tools.
 
 Creating a new application
 --------------------------
 
-Let's start by creating a simple Haskell application from scratch where we'll
-learn about a Haskell package's directory structure, how to run the executable,
-and how to add external dependencies.
+We create a minimal Haskell application to get a quick overview 
+about the ``cabal`` commandline tool:
 
+1. How to initialize a Haskell package.
+2. How files are organized inside a package.
+3. How to compile Haskell files and run a resulting executable.
+4. How to manage external dependencies.
 
-Initializing the application
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Initializing an application
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Start by initialising our ``myfirstapp`` project, these instructions work in
-unix shells and PowerShell (if you're on Windows).
+To initialize a new Haskell application, run
 
 .. code-block:: console
 
-    $ cabal init myfirstapp -n
-
-.. note:: ``myfirstapp`` stands for the directory (or path) where the project
-          will reside in, if omitted, ``cabal init`` will do its proceedings
-          in the directory it's called in.
-
-.. note:: ``-n`` stands for ``--non-interactive``, which means that cabal will try to guess
-          how to set up the project for you and use the default settings, which will serve us
-          well for the purpose of this tutorial.
-          When setting up your projects in the future, you will likely want to omit ``-n``
-          and do just ``cabal init``, so that cabal will interactively ask you
-          for the details on how the project should be set up
-          (while still offering reasonable defaults on every step).
-          Also, you can run ``cabal init --help`` to get more info on how ``cabal init`` can be used.
+    $ cabal init myapp --non-interactive
 
-This will generate the following files:
+in a terminal. This generates the following files in a new ``myapp`` folder:
 
 .. code-block:: console
 
     $ tree
     .
-    └── myfirstapp
+    └── myapp
         ├── app
         │   └── Main.hs
         ├── CHANGELOG.md
-        └── myfirstapp.cabal
+        └── myapp.cabal
 
-``app/Main.hs`` is where your package's code lives.
+``myapp.cabal`` is Cabal's metadata file which describes your package,
+its dependencies and how it is built:
```

Comment:
```suggestion
its dependencies, and how it is built:
```
oxford commas FTW

Reply:
Actually, I think it'd be good to be a little more precise here and say that a package description contains package metadata and a sequence of sections, each defining a _package component_. Components can be of one of several types (executables, libraries, etc.). Every component describes, among other things, the location of source files and its dependencies. 

Could you integrate this description into the text?

Reply:
Done